File: Copilot/3.Desarrollo Web Endpoint RESTful/java/src/Singleton.java
public class Singleton {
    // Instancia privada y estática
    private static Singleton instance;

    // Constructor privado para evitar instanciación externa
    private Singleton() {
        System.out.println("Instancia creada");
    }

    // Método público y estático para obtener la instancia
    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }

    // Método de prueba
    public static void main(String[] args) {
        Singleton a = Singleton.getInstance();
        Singleton b = Singleton.getInstance();

        System.out.println("¿Es la misma instancia? " + (a == b));
    }
}

File: Copilot/1. Algoritmo Básico Ordenamiento por Burbuja/singleton.ts
class Singleton {
    private static instance: Singleton;

    private constructor() {
        console.log("Instancia creada");
    }

    static getInstance(): Singleton {
        if (!Singleton.instance) {
            Singleton.instance = new Singleton();
        }
        return Singleton.instance;
    }

    saludar(): void {
        console.log("Hola desde Singleton");
    }
}

// Prueba
const instancia1 = Singleton.getInstance();
const instancia2 = Singleton.getInstance();

console.log("¿Son la misma instancia?", instancia1 === instancia2); // true
instancia1.saludar();
